components/bottom-nav/Android/Android-bottom-nav.md

```diff
@@ -17,155 +16,140 @@ path: /components/bottom-navigation
- </selector>
+      ...
+      android:title="@string/text_label"/>
+  ...
+</menu>
 ```
 
-And you would use it like this on your `BottomNavigationView`:
+The `labelVisibilityMode` attribute can be used to adjust the behavior of the text labels for each navigation item. There are four visibility modes:
 
-```xml
-<com.google.android.material.bottomnavigation.BottomNavigationView
-    android:id="@+id/bottom_navigation"
-    android:layout_width="match_parent"
-    android:layout_height="wrap_content"
-    style="@style/Widget.MaterialComponents.BottomNavigationView"
-    app:menu="@menu/bottom_navigation_menu" />
-```
-
-### Material Styles
+* `LABEL_VISIBILITY_AUTO` (default): The label behaves as “labeled” when there are 3 items or less, or “selected” when there are 4 items or more
+* `LABEL_VISIBILITY_SELECTED`: The label is only shown on the selected navigation item
+* `LABEL_VISIBILITY_LABELED`: The label is shown on all navigation items
+* `LABEL_VISIBILITY_UNLABELED`: The label is hidden for all navigation items
 
-Using `BottomNavigationView` with an updated Material theme
-(`Theme.MaterialComponents`) will provide the correct updated Material styles to
-your `BottomNavigationView`s by default. If you need to use an updated
-`BottomNavigationView` and your application theme does not inherit from an
-updated Material theme, or if you'd like to use a variant style, you can apply
-one of the updated Material styles directly to your widget in XML:
+### Adding badges
 
-#### Updated Material Style (Default)
+![Bottom navigation with 3 icons with badges, an icon only badge and two numbered badges showing 99 and 999+](assets/bottom-navigation-badges.png)
 
-The default Material `BottomNavigationView` style consists of updated colors,
-text sizing, and behavior. The default BottomNavigationView has white background
-and icons and text colored with `colorPrimary`.
+Initialize and show a `BadgeDrawable` associated with `menuItemId`, subsequent calls to this method will reuse the existing `BadgeDrawable`:
 
-```
-style="@style/Widget.MaterialComponents.BottomNavigationView"
+```kt
+var badge = bottomNavigation.getOrCreateBadge(menuItemId)
+badge.isVisible = true
+// An icon only badge will be displayed unless a number is set:
+badge.number = 99
 ```
 
-#### Colored Material Style
+Remove any `BadgeDrawable`s that are no longer needed:
 
-This style inherits from the default style but sets the colors to different
-mappings. Use the colored style to get a bottom navigation bar with a
-`colorPrimary` background and shades of white for the icon and text colors.
-
-```
-style="@style/Widget.MaterialComponents.BottomNavigationView.Colored"
+```kt
+bottomNavigation.removeBadge(menuItemId)
 ```
 
-#### Legacy Style
+Or, as best practice, if you only need to temporarily hide the badge (e.g. until the next notification is received), the recommended/lightweight alternative is to change the visibility of the BadgeDrawable instead:
```

Comment:
Suggested change:

"As best practice if you need to temporarily hide the badge (e.g. until the next notification is received), change the visibility of `BadgeDrawable`:"